File: baekjoon-online-judge/gold/11758.java
import java.io.*;
import java.util.*;

public class Main {
	static class Point {
		public final long x, y;
		Point(long x, long y) {
			this.x = x;
			this.y = y;
		}
	}

	private static int ccw(Point a, Point b, Point c) {
		long abx = b.x - a.x;
		long aby = b.y - a.y;
		long acx = c.x - a.x;
		long acy = c.y - a.y;

		long cross = abx * acy - aby * acx;
		if (cross > 0) return 1;
		if (cross < 0) return -1;
		return 0;
	}

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		// Input
		List<Point> list = new ArrayList<>();
		for (int i = 0; i < 3; ++i) {
			String[] in = br.readLine().split(" ");
			int X = Integer.parseInt(in[0]);
			int Y = Integer.parseInt(in[1]);
			list.add(new Point(X, Y));
		}

		// Solve & Output
		System.out.println(ccw(list.get(0), list.get(1), list.get(2)));
	}
}

File: baekjoon-online-judge/gold/11758.cpp
#include <iostream>

using namespace std;

typedef struct __point {
    int x, y;
} Point;

int CCW(const Point& A, const Point& B, const Point& C)
{
    int cross_product = (B.x - A.x)*(C.y - A.y) - (C.x - A.x)*(B.y - A.y);
    if (!cross_product) return 0;
    return cross_product > 0 ? 1 : -1;
}

int main()
{
    Point P[3];
    for (int i = 0; i < 3; ++i) cin >> P[i].x >> P[i].y;
    cout << CCW(P[0], P[1], P[2]);
}
